ops/internal/deployer/opaque_map.go

```diff
@@ -0,0 +1,140 @@
+			if val == float64(int64(val)) {
+				m[k] = int64(val)
+			}
+		case map[string]any:
+			// Recursively process nested maps
+			useInts(val)
+		case []any:
+			// Process arrays
+			for i, item := range val {
+				if fItem, ok := item.(float64); ok && fItem == float64(int64(fItem)) {
+					val[i] = int64(fItem)
+				} else if mapItem, ok := item.(map[string]any); ok {
+					useInts(mapItem)
+				}
+			}
+		}
+	}
+}
+
+func (om OpaqueState) ReadL1ChainID() (uint64, error) {
+	node := dasel.New(om)
+	l1ChainIDNode, err := node.Query("appliedIntent.l1ChainID")
+	if err != nil {
+		return 0, fmt.Errorf("failed to read L1 chain ID: %w", err)
+	}
+	l1ChainIDFloat, ok := l1ChainIDNode.InterfaceValue().(float64)
+	if !ok {
+		return 0, errors.New("failed to parse L1 chain ID")
+	}
+	return uint64(l1ChainIDFloat), nil
+}
+
+func (om OpaqueState) ReadL1ContractsLocator() (string, error) {
+	node := dasel.New(om)
+	l1ContractsReleaseNode, err := node.Query("appliedIntent.l1ContractsLocator")
+	if err != nil {
+		return "", fmt.Errorf("failed to read L1 contracts release: %w", err)
+	}
+	l1ContractsRelease, ok := l1ContractsReleaseNode.InterfaceValue().(string)
+	if !ok {
+		return "", errors.New("failed to parse L1 contracts release")
+	}
+	return l1ContractsRelease, nil
+}
+
+func (om OpaqueState) ReadL2ContractsLocator() (string, error) {
+	node := dasel.New(om)
+	l2ContractsReleaseNode, err := node.Query("appliedIntent.l2ContractsLocator")
+	if err != nil {
+		return "", fmt.Errorf("failed to read L2 contracts release: %w", err)
+	}
+	l2ContractsRelease, ok := l2ContractsReleaseNode.InterfaceValue().(string)
+	if !ok {
+		return "", errors.New("failed to parse L2 contracts release")
+	}
+	return l2ContractsRelease, nil
+}
+
+func (om OpaqueState) ReadL2ChainId(idx int) (string, error) {
+	node := dasel.New(om)
+	l2ChainIdNode, err := node.Query(fmt.Sprintf("appliedIntent.chains.[%d].id", idx))
+	if err != nil {
+		return "", fmt.Errorf("failed to read L2 chain ID: %w", err)
+	}
+	l2ChainId, ok := l2ChainIdNode.InterfaceValue().(string)
+	if !ok {
+		return "", errors.New("failed to parse L2 chain ID")
+	}
+	return l2ChainId, nil
+}
+
+func (om OpaqueState) ReadSystemConfigProxy(idx int) (common.Address, error) {
+	node := dasel.New(om)
+	systemConfigProxyNode, err := node.Query(fmt.Sprintf("appliedIntent.opChainDeployments.[%d].SystemConfigProxy", idx))
+	if err == nil {
+		systemConfigProxy, ok := systemConfigProxyNode.InterfaceValue().(string)
+		if !ok {
+			return common.Address{}, errors.New("failed to parse SystemConfigProxy")
+		}
+		return common.HexToAddress(systemConfigProxy), nil
+	}
+
+	// Fallback to the legacy field name
+	systemConfigProxyNode, err = node.Query(fmt.Sprintf("appliedIntent.opChainDeployments.[%d].systemConfigProxyAddress", idx))
+	if err != nil {
+		return common.Address{}, fmt.Errorf("failed to read SystemConfigProxy: %w", err)
+	}
+
+	systemConfigProxy, ok := systemConfigProxyNode.InterfaceValue().(string)
```

Comment:
[nit] we may want a utility method that gets a string from a node, since we're duplicating a lot of code here to get strings out

Reply:
Done in https://github.com/ethereum-optimism/superchain-registry/pull/1041/commits/1fba7e052bf9101c629c66b62fe5ee4e66b0c02d